Makefile

```diff
@@ -226,7 +226,7 @@ $$(build_private_libdir)/sys$1.o: $$(build_private_libdir)/inference.ji $$(JULIA
 		mv $$@.tmp $$@; \
 	else \
 		echo '*** This error is usually fixed by running `make clean`. If the error persists$$(COMMA) try `make cleanall`. ***' && false; \
-	fi )
+	fi && $$(call spawn,$3) $$(JULIAHOME)/etc/write_base_cache.jl)
```

Comment:
Separate this out into it's own build step.

```
julia-base-cache: julia-sysimg-% | $(DIRS) $(build_datarootdir)/julia/base
    (call exec,$(JULIA_EXECUTABLE) $$JULIAHOME)/etc/write_base_cache.jl $(build_datarootdir)/julia/base/base.cache)
```

You might want to actually let make decide the location of the cache file and pass that to the script instead of letting the script discover if we are doing an out-of-source build, then you can add it to the list of dependencies for in `L114` for `julia-release julia-debug`.

At least that's how I would start.

Reply:
You saved me a bunch of time. Thanks so much.

Reply:
Though do you really think it should go in `usr/share/julia/base`? For me that's a symlink, so it actually ends up `base/`, which is why I ended up adding a `.gitignore`. What about just putting it in `usr/share/julia` directly?

Reply:
That should be OK as well. If you want you can test out-of-tree builds and